src/nvim/viml/parser/expressions.c

```diff
@@ -2330,15 +2329,14 @@ ExprAST viml_pexpr_parse(ParserState *const pstate, const int flags)
             break;
           } else if (eastnode_type == kExprNodeSubscript) {
             is_subscript = true;
-            can_be_ternary = false;
             assert(!is_ternary);
             break;
```

Comment:
clang warns about this case because of the `break`. The other cases shouldn't be removed, because they are in a for-loop.